js/src/traceable.ts

```diff
@@ -165,6 +165,15 @@ export function traceable<Func extends (...args: any[]) => any>(
     return new Promise((resolve, reject) => {
       void asyncLocalStorage.run(currentRunTree, async () => {
         try {
+          const onEnd = config?.on_end;
+          if (onEnd) {
+            if (!currentRunTree) {
+              throw new Error(
+                "Can not call 'on_end' if currentRunTree is undefined"
```

Comment:
hm - we shouldn't ever break someone's execution flow from tracing

Reply:
Switched to a console.error